src/strands/tools/mcp/mcp_from_config.py

```diff
@@ -0,0 +1,228 @@
+        """Set transport type after initialization."""
+        self.transport_type = MCPTransportType.SSE
+
+    def create_transport_callable(self) -> Callable[[], MCPTransport]:
+        """Create SSE transport callable."""
+
+        def transport_callable() -> MCPTransport:
+            headers = self.headers.copy() if self.headers else {}
+            if self.authorization_token:
+                headers["Authorization"] = f"Bearer {self.authorization_token}"
+            return sse_client(self.url, headers=headers if headers else None)
+
+        return transport_callable
+
+
+def _infer_transport_from_url(url: str) -> str:
+    """Infer transport type from URL when none specified.
+
+    - If path contains '/sse' (optionally followed by / ? & or end), treat as 'sse'
+    - Otherwise default to 'streamable-http'
+    """
+    try:
+        path = (urlparse(url).path or "").lower()
+    except Exception:
+        return "streamable-http"
+    return "sse" if re.search(r"/sse(/|\?|&|$)", path) else "streamable-http"
+
+
+class MCPServerConfig:
+    """Configuration for an MCP server following MCP standards."""
+
+    def __init__(self, transport_config: MCPTransportConfig) -> None:
+        """Initialize MCP server configuration."""
+        self.transport_config = transport_config
+        self.name = transport_config.name
+        self.timeout = transport_config.timeout
+
+    def create_client(self) -> MCPClient:
+        """Create an MCPClient from this configuration."""
+        return MCPClient(self.transport_config.create_transport_callable())
+
+    @classmethod
+    def from_config(cls, config_path: str) -> List["MCPServerConfig"]:
+        """Load MCP server configurations from a config file."""
+        config_file = Path(config_path)
+        if not config_file.exists():
+            raise FileNotFoundError(f"Config file not found: {config_path}")
+
+        with open(config_file) as f:
+            config_data = json.load(f)
+
+        if "mcpServers" in config_data and config_data["mcpServers"] is not None:
+            return cls._parse_mcp_servers_format(config_data["mcpServers"])
+
+        return []
+
+    @classmethod
+    def _parse_mcp_servers_format(cls, mcp_servers: Dict[str, Dict]) -> List["MCPServerConfig"]:
+        """Parse mcpServers format (Claude Desktop/legacy and enhanced)."""
+        servers = []
+        server_names = set()
+
+        for name, server_config in mcp_servers.items():
+            if len(name) == 0 or len(name) > 250 or name in server_names:
+                raise ValueError(f"Invalid server name: {name}")
+            server_names.add(name)
+
+            # Accept multiple keys, infer from URL when absent
+            raw_transport = (
+                server_config.get("transport")
+                or server_config.get("transportType")
+                or server_config.get("transport_type")
+            )
+            if raw_transport is None and "url" in server_config:
```

Comment:
Are there any other config formats that support streamable http or sse?  If so, do they use the same logic for deciding on the format?  If not, let's make it a hard requirement that you specify the transport type for non stdio formats. I'd rather be safe than sorry.